pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/reader/JacksonJsonReader.java

```diff
@@ -33,17 +34,17 @@
  */
 public class JacksonJsonReader<T> implements SchemaReader<T> {
     private final Class<T> pojo;
-    private final ObjectMapper objectMapper;
+    private final ObjectReader objectReader;
 
     public JacksonJsonReader(ObjectMapper objectMapper, Class<T> pojo) {
         this.pojo = pojo;
-        this.objectMapper = objectMapper;
+        this.objectReader = pojo != null ? objectMapper.readerFor(pojo) : objectMapper.reader();
```

Comment:
null handling is done this way because of backwards compatibility. The previous solution accepted null as pojo and a test breaks if this logic doesn't exist for the ObjectReader.